realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -1277,41 +1277,82 @@ private static boolean isClassForRealmModel(Class<?> clazz) {
         this.osList = osList;
     }
 
-    abstract boolean forRealmModel();
+    public abstract boolean forRealmModel();
 
-    OsList getOsList() {
+    public final OsList getOsList() {
         return osList;
     }
 
-    final boolean isValid() {
+    public final boolean isValid() {
         return osList.isValid();
     }
 
-    final int size() {
+    public final int size() {
         final long actualSize = osList.size();
         return actualSize < Integer.MAX_VALUE ? (int) actualSize : Integer.MAX_VALUE;
     }
 
-    final boolean isEmpty() {
+    public final boolean isEmpty() {
         return osList.isEmpty();
     }
 
-    final void checkInsertIndex(int index) {
-        final int size = size();
-        if (index < 0 || size < index) {
-            throw new IndexOutOfBoundsException("Invalid index " + index + ", size is " + osList.size());
+    protected abstract void checkValidValue(@Nullable Object value);
+
+    @Nullable
+    public abstract T get(int index);
+
+    public final void append(@Nullable Object value) {
```

Comment:
Now the question is, what is the difference between `List.add()` and `RealmList.append()`.